webrender/src/prim_store.rs

```diff
@@ -611,12 +614,11 @@ impl PrimitiveStore {
                 }
                 PrimitiveKind::Image => {
                     let image_cpu = &mut self.cpu_images[metadata.cpu_prim_index.0];
-                    // If the image primitive uses externally allocated textures, it
-                    // is already resolved and doesn't not use the resource cache.
-                    // TODO(nical): choose whether to use the source texture or the kind
-                    // to differentiate between external and regular images, and stick to
-                    // it more concistently.
-                    if !image_cpu.color_texture_id.is_external() {
+                    if let Some(key) = image_cpu.color_texture_id.to_external() {
+                        // If the image primitive uses externally allocated textures,
+                        // we resolve it on the renderer.
+                        self.deferred_image_primitives.push((key, metadata.gpu_prim_index));
```

Comment:
Do you ever force the primitive to appear in the `prims_to_resolve` list? Perhaps, it would be easier to straight push to `deferred_image_primitives` from `prepare_prim_for_render()`.